srcalc/src/main/webapp/WEB-INF/views/editProcedures.jsp

```diff
@@ -0,0 +1,43 @@
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/functions" prefix="fn" %>
+<%@ taglib tagdir="/WEB-INF/tags" prefix="srcalc" %>
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
+
+<srcalc:adminPage title="Update Procedure Set">
+    <h2>Update Procedure Set</h2>
+    <p>There are currently ${fn:length(procedures)} procedures in the database.</p>
+    
+    <h3>Upload a New Procedure Set</h3>
+    <c:if test="${uploadSuccess}">
+    <p class="successMessage">The new procedure set has been saved to the database.</p>
+    </c:if>
+    <c:if test="${!empty validationErrors}">
+    <div class="error">
+    <p>The uploaded procedure set was not valid. The database has not been modified.</p>
+    <p>Errors were:</p>
+    <ul>
+        <c:forEach var="error" items="${validationErrors}">
+        <li>${error.locationPrefix}<spring:message message="${error}" /></li>
+        </c:forEach>
+    </ul>
```

Comment:
I feel like the errors should be after the upload button so that the button is easier to find than having to scroll to the bottom of the page.